numba/np/arraymath.py

```diff
@@ -4040,15 +4041,18 @@ def histogram_impl(a, bins=10, range=None):
 
 iinfo = namedtuple('iinfo', _iinfo_supported)
 
+# Suppress deprecation warnings while keeping support MachAr
+with warnings.catch_warnings(record=True):
+    warnings.simplefilter("ignore", DeprecationWarning)
 
-@overload(np.MachAr)
-def MachAr_impl():
-    f = np.MachAr()
-    _mach_ar_data = tuple([getattr(f, x) for x in _mach_ar_supported])
+    @overload(np.MachAr)
+    def MachAr_impl():
+        f = np.MachAr()
+        _mach_ar_data = tuple([getattr(f, x) for x in _mach_ar_supported])
 
-    def impl():
-        return MachAr(*_mach_ar_data)
-    return impl
+        def impl():
+            return MachAr(*_mach_ar_data)
+        return impl
```

Comment:
It'll be better to minimize the amount of code captured by `catch_warnings`. One way to do it is to only have `np_MachAr = np.MachAr` inside it. That way we don't accidentally suppress warnings from other code; like in `@overload(...)`

Reply:
The overload also needs to raise the same `DeprecationWarning` for those using `np.MachAr` inside `@jit`